WcaOnRails/app/models/abstract_cached_model.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+class AbstractCachedModel < ActiveRecord::Base
```

Comment:
Thanks for the interesting reads!
I'll look into reworking the cache mechanism this evening or tomorrow.

As for caching the `.all`: this is for sure *not* a critical part of the performances, however we do `Country.all` or `Country.real` (scoped `all`) at several places, so in the case where we don't catch new records we can avoid these queries "for free".